### PR TITLE
misc

### PR DESCRIPTION
**Pull request by oliver-sanders**
Misceallaneous stuff from recent reviews...

terminology: workflow-name => workflow-id                                                                                          
    
* Workflow name still exists as an install concept in relation to the source    
  directory, and in the run directory where it is the workflow ID with    
  the run name removed.    
    
terminology: reg->id                               
                                                            
* Reg was a Cylc <= 6 concept relating to the registration DB.    
* Beyond Cylc 6 we didn't have a better name for this.    
* With the universal ID, workflow reg became workflow ID.    
    
tests/integration: set run_mode=simulation as the default    
    
* Integration tests shouldn't go running workflows for real.    
* Set the simulation run mode as the default, this avoids shell    
  interaction and reduces the number of moving parts.  

config: fill small coverage gap

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Review by wxtim — APPROVED** [review]
Looks good. Mostly simple, but with the test changes needs a second review IMO.

File: cylc/flow/scripts/install.py
Comment (MetRonnie) [inline cylc/flow/scripts/install.py:30]: Not this one I think

```suggestion
the first match. The installed workflow name will be the same as SOURCE_NAME,
```
or perhaps
```suggestion
the first match. The installed workflow ID will start with SOURCE_NAME,
```

File: cylc/flow/pathutil.py
Comment (MetRonnie) [inline cylc/flow/pathutil.py:160]:
```suggestion
        named_sub_dir: e.g workflow_name/run1
```

File: cylc/flow/pathutil.py
Comment (MetRonnie) [inline cylc/flow/pathutil.py:207]:
```suggestion
        workflow_id: full id of the run, e.g. myflow/run1
```

File: cylc/flow/install.py
Comment (MetRonnie) [inline cylc/flow/install.py:587]:
```suggestion
        WorkflowFilesError if run_dir is nested inside an existing run dir,
            or install dirs are nested.
```

File: cylc/flow/install.py
Comment (oliver-sanders) [reply cylc/flow/install.py:551]: Ah right, I'll change the arg description to avoid "workflow directory".

File: cylc/flow/install.py
Comment (oliver-sanders) [inline cylc/flow/install.py:550]:
```suggestion
        run_path_base: Absolute path of the parent of the workflow's run dir,
```

**Pull request merged**
